python/kvikio/cufile.py

```diff
@@ -197,13 +196,13 @@ def read(self, buf, size: int = None, file_offset: int = 0, task_size=None) -> i
 
         Notes
         -----
-        cuFile can only make use of GPUDirect Storage for reads that are
-        aligned to a page boundary. The GPU page size used by cuFile is 4kB, so
-        all reads must be at an offset that is a multiple of 4096 bytes. If the
-        desired `file_offset` is not a multiple of 4096 it is likely desirable
-        to round down to the nearest multiple of 4096 and discard any undesired
-        bytes from the resulting data. Similarly, it is optimal for `size` to
-        be a multiple of 4096 bytes.
+        When reading into device memory, cuFile can only make use of GDS for reads
```

Comment:
Same thought as above except reading into pinned memory and copying to device memory.